filebeat/input/tcp/conn.go

```diff
@@ -0,0 +1,72 @@
+package tcp
+
+import (
+	"io"
+	"net"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// ErrMaxReadBuffer returns when too many bytes was read on the io.Reader
+var ErrMaxReadBuffer = errors.New("max read buffer reached")
+
+// MeteredReader is based on LimitedReader but allow to reset the byte read and return a specific
+// error when we reach the limit.
+type MeteredReader struct {
```

Comment:
I am not a big fan of this name. It would be nice to express that it is a special `LimitedReader`.
However, I am not able to come up with a proper name. My best alternative proposal is `ResetterLimitedReader`. I am willing to let the change go.

Reply:
Naming is hard but, ResetterLimitedReader or ResetableLimitedReader. I am ok with both.